copy/single.go

```diff
@@ -424,6 +452,7 @@ func (ic *imageCopier) copyLayers(ctx context.Context) error {
 			}
 		} else {
 			cld.destInfo, cld.diffID, cld.err = ic.copyLayer(ctx, srcLayer, toEncrypt, pool, index, srcRef, manifestLayerInfos[index].EmptyLayer)
+			res = append(res, *cld.destInfo.CompressionAlgorithm)
```

Comment:
This runs in a concurrently-executing goroutine, so it is invalid to modify a shared `res` variable from here. Instead, pass it through a new field in `copyLayerData`, and collect (+deduplicate?) it when processing the goroutine results.